src/pages/travel/visa/index.tsx

```diff
@@ -110,70 +153,68 @@ const VisaPage: React.FC = () => {
     }
   }, [searchTerm, allCountries]);
 
-  const handleSelectPickerChange = (
-    option: { label: string; value: string } | null
-  ) => {
-    if (option) {
-      checkVisaRequirement(option.value);
-    } else {
-      setSelectedCountry(null);
-      setVisaRequirement(null);
-    }
-  };
-
   const checkVisaRequirement = (country: string) => {
     setSelectedCountry(country);
-
-    const typedVisaData = visaData as PhilippinesVisaPolicy;
-
-    // Check if country is in 30-day visa-free list
-    if (typedVisaData.visaFreeEntryPolicies[0].countries?.includes(country)) {
-      setVisaRequirement({
-        type: 'visa-free',
-        duration: '30 days',
-        description: typedVisaData.visaFreeEntryPolicies[0].description,
-      });
-      return;
+    const requirement = countryRequirements.get(country);
+    if (requirement) {
+      setVisaRequirement(requirement);
     }
+  };
 
-    // Check if country is in 59-day visa-free list
-    if (typedVisaData.visaFreeEntryPolicies[1].countries?.includes(country)) {
-      setVisaRequirement({
-        type: 'visa-free',
-        duration: '59 days',
-        description: typedVisaData.visaFreeEntryPolicies[1].description,
-      });
-      return;
+  const openDetailsDialog = (country: string) => {
+    setDialogCountry(country);
+    const requirement = countryRequirements.get(country);
+    if (requirement) {
+      setVisaRequirement(requirement);
     }
+    setDialogOpen(true);
+  };
```

Comment:
i found an edge case for the grid view/detail view interaction:

when we're on detail view and we select a country, the url updates to below for example:
> http://localhost:5173/travel/visa?country=Andorra&view=detail

but when we switch back to grid view, its still there - just clarifying if this is intended behavior?

Reply:
It's supposedly intended so they don't have to search again when changing views. The heirarchy of the search is intentionally higher than both views to they can re-use it.